kafka/client.go

```diff
@@ -436,58 +454,57 @@ func isPartitionRFChanging(status *sarama.PartitionReplicaReassignmentsStatus) b
 	return len(status.AddingReplicas) != 0 || len(status.RemovingReplicas) != 0
 }
 
-func (client *Client) ReadTopic(name string) (Topic, error) {
+func (client *Client) ReadTopic(name string, refresh_metadata bool) (Topic, error) {
 	c := client.client
 
 	topic := Topic{
 		Name: name,
 	}
 
-	err := c.RefreshMetadata()
-	if err != nil {
-		log.Printf("[ERROR] Error refreshing metadata %s", err)
-		return topic, err
+	if refresh_metadata {
+		log.Printf("[DEBUG] Refreshing metadata");
+		err := c.RefreshMetadata()
+		if err != nil {
+			log.Printf("[ERROR] Error refreshing metadata %s", err)
+			return topic, err
+		}
+		err = client.extractTopics()
+		if err != nil {
+			return topic, err
+		}
+	} else {
+		log.Printf("[DEBUG] skipping metadata refresh")
 	}
-	topics, err := c.Topics()
 
-	if err != nil {
-		log.Printf("[ERROR] Error getting topics %s from Kafka", err)
-		return topic, err
-	}
+	if _, ok := client.topics[name]; ok {
```

Comment:
rather than an O(n) search through all the topics, just do an O(1) search through the cached map.